helix-core/src/main/java/org/apache/helix/ConfigAccessor.java

```diff
@@ -87,11 +88,11 @@
    * @param builder
    */
   private ConfigAccessor(Builder builder) {
-    switch (builder._realmMode) {
+    switch (builder.getRealmMode()) {
```

Comment:
Isn't this switch block identical to the ones in ZKHelixAdmin and ClusterSetup? IMO, let the builder have a method for creating a ZkClient is simpler and easier to maintain.

Reply:
Moved all construction logic to the abstract builder, and refactored the code out of these Java API classes.